### PR TITLE
feat(variables): add view at project & environment levels

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1046

### CODE REVIEW

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  mode: 'CREATE' | 'UPDATE'
+  type: keyof typeof APIVariableTypeEnum
+  isFile?: boolean
+}
+
+export function CreateUpdateVariableModal({
+  parentId,
+  scope,
+  closeModal,
+  onSubmit: _onSubmit,
```
Comment (RemiBonnet) [inline libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx:41]: Should be the inverse, same you have done on the `VariablesActionToolbar` component.
_onSubmit is externe

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+  variable,
+  mode,
+  type,
+  isFile: _isFile,
```
Comment (RemiBonnet) [inline libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx:45]: Same here

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa ! Below some improvements I seen, in addition to creation bugs we had discuss during call

- Is it possible to fix or find another solution with this gap between button and separator?
https://www.loom.com/share/cde524182d984bc09ca8e19ef420ca84

- The left column is little bit huge, can you reduce it ? I don't find design about it
Screenshot 2024-04-29 at 10 53 10

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx]: https://github.com/Qovery/console/pull/1318/commits/b9aa846a31eb056e479de8bbc75c408d2154dba7

File: libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx]: https://github.com/Qovery/console/pull/1318/commits/b9aa846a31eb056e479de8bbc75c408d2154dba7

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**